service/src/test/java/com/ge/predix/acs/privilege/management/dao/GraphResourceRepositoryTest.java

```diff
@@ -528,33 +518,53 @@ private ResourceEntity persistResourceWithParentsToZoneAndAssert(final ZoneEntit
         resource.setAttributes(attributes);
         resource.setAttributesAsJson(JSON_UTILS.serialize(resource.getAttributes()));
         resource.setParents(parents);
-        ResourceEntity resourceEntity = this.resourceRepository.save(resource);
+        ResourceEntity resourceEntity = concurrentSave(resource, 0);
         assertThat(this.resourceRepository.findOne(resourceEntity.getId()), equalTo(resource));
         return resourceEntity;
     }
 
-    private ResourceEntity persistResource0toZone1AndAssert() {
-        return persistResourceToZoneAndAssert(TEST_ZONE_1, BASEMENT_SITE_ID, BASEMENT_ATTRIBUTES);
+    private ResourceEntity persistRandomResourcetoZone1AndAssert() {
+        return persistResourceToZoneAndAssert(TEST_ZONE_1, DRIVE_ID + getRandomNumber(), DRIVE_ATTRIBUTES);
     }
 
-    private ResourceEntity persistResource1toZone1AndAssert() {
-        return persistResourceToZoneAndAssert(TEST_ZONE_1, DRIVE_ID, DRIVE_ATTRIBUTES);
+    private ResourceEntity persistResource0toZone1AndAssert() {
+        return persistResourceToZoneAndAssert(TEST_ZONE_1, BASEMENT_SITE_ID, BASEMENT_ATTRIBUTES);
     }
 
     private ResourceEntity persistResource2toZone1AndAssert() {
         return persistResourceToZoneAndAssert(TEST_ZONE_1, JOSECHUNG_ID, DRIVE_ATTRIBUTES);
     }
 
-    private ResourceEntity persistResource1toZone2AndAssert() {
-        return persistResourceToZoneAndAssert(TEST_ZONE_2, DRIVE_ID, DRIVE_ATTRIBUTES);
+    private int getRandomNumber() {
+        return randomGenerator.nextInt(10000000);
+    }
+
+    private ResourceEntity concurrentSave(ResourceEntity resource, int retryTimes) throws TitanException {
```

Comment:
- rename to saveWithRetry ?  
- rename retryTimes to retryCount. honor retryCount by counting down to 0
- also  this meeting can be shared between Resource and Subject tests.  (use the base class ZonableEntity)